### DISCUSSION TITLE
Does anyone else wish for a Thonny "portable" for MacOS?

### DISCUSSION THREAD

**Discussion by villares** [thonny/thonny#3090] [General]
I have been using the Windows Thonny IDE portable, adding some plug-ins and libraries, to simplify the on-boarding of my students.

Now I have many MacOS students and, due to no fault of Thonny, but due to the complexities of the thonny-py5mode plug-in and py5 library, they have struggled with the installation. It would help me a lot it there were a Thonny bundle I could tweak and they could just copy from a .dmg image on Mac...

Someone mentioned py2app (https://py2app.readthedocs.io/en/latest/index.html) to me, but I'm not sure one could "build" Thonny using it, or if "signing" or other issues would make it impossible.

**Comment by aivarannamaa** [comment]
Have you looked at `packaging/mac/create_release.sh` in Thonny's code? This is what I'm using to create the pkg-installers. The required context of running this script:

* You have installed a suitable Python from python.org
* You have made a copy of it under your home directory with `packaging/mac/prepare_base_bundle.sh` (this script also prepares the structure of Thonny application bundle)
* You have my code signing sertificates set up in your computer

The last item is problematic, of course, but you can skip signing, if you comment out line 12 in create_release.sh. Also, feel free to prescribe a specific Thonny version on line 5.

If you skip sigining, then you probably need to tweak `packaging/mac/scripts/postinstall` so that it [removes quarantine bits from all installed files](https://superuser.com/a/1336045/48092) as the last step of installation. You would probably also want to update files in `packaging/mac/resource_templates`. These tweaks need to be done before running `packaging/mac/prepare_base_bundle.sh`, as this scripts copies these files to the preparation folder.

This explains how you could create your own pkg for plain Thonny. If you want to add Py5 stuff, then you may want to also tweak `packaging/requirements-regular-bundle.txt` and maybe add something manually to `~/thonny_template_build_312`.

In the master-branch, the files mentioned above are meant for bundling Thonny 5 (not released yet) with Python 3.12. For older Thonny and Python versions you may be better off with branch `v4_1`.

**Reply by aivarannamaa** [reply]
> you can skip signing,

Of course, you don't need to. If you have your own code signing certificates (need to pay Apple 99$ per year for this), then you can simply update the identifiers in `packaging/mac/sign_bundle_in_build.sh`. Then you could also [notarize](https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution) the pkg, which would make the installation process easier for your students.

**Reply by villares** [reply]
Thank you so much for Thonny, and thanks for these wonderfully detailed instructions. I don't have a Mac, but I might have access to one, or ideally, I might find a friend  with MacOS developer signing certificates to help...
